rodan-main/code/rodan/jobs/MEI_encoding/build_mei_file.py

```diff
@@ -586,15 +635,48 @@ def build_mei(pairs: List[Tuple[List[dict], dict]], classifier: dict, width_cont
 
                 continue
 
+
             cur_staff = int(glyph['staff'])
 
+            # if multi column and next system is new column, add new cb
+            if is_multi_column:
+                if staff_to_column[cur_staff] > prev_column:
+                    # add cb to layer
+                    zoneId = generate_zone(surface, previous_cb["bb"])
+                    previous_cb["cb"].set("facs", "#" + zoneId)
+
+                    #start new cb
+                    prev_column = staff_to_column[cur_staff]
+                    cb = new_el("cb")
+                    cb.set("n", str(staff_to_column[cur_staff] + 1))
+                    previous_cb["cb"] = cb
+                    previous_cb["bb"] = None
+                    layer.append(cb)
+
             bb = staves[cur_staff]['bounding_box']
+            # if there are multiple columns, shift the bounding box accordingly
             bb = {
                 'ulx': bb['ulx'],
                 'uly': bb['uly'],
                 'lrx': bb['ulx'] + bb['ncols'],
                 'lry': bb['uly'] + bb['nrows'],
             }
+
+            # if multi column move the bounding box back
+            # adjust current cb zone to include newest sb bb
+            if is_multi_column:
+                bb = translate_bbox(bb, column_split_info["split_ranges"], height, staff_to_column[cur_staff])
+                if previous_cb["bb"] is None:
+                    previous_cb["bb"] = bb
+                else:
+                    if previous_cb["bb"]["ulx"] > bb["ulx"]:
+                        previous_cb["bb"]["ulx"] = bb["ulx"]
+                    if previous_cb["bb"]["uly"] > bb["uly"]:
+                        previous_cb["bb"]["uly"] = bb["uly"]
+                    if previous_cb["bb"]["lrx"] < bb["lrx"]:
+                        previous_cb["bb"]["lrx"] = bb["lrx"]
+                    if previous_cb["bb"]["lry"] < bb["lry"]:
+                        previous_cb["bb"]["lry"] = bb["lry"]

```

Comment:
a lot of these `if is_multi_column` bits get distantly indented from the left side (think this one hits six tabs). is there some way you can refactor this so that the multi column logic is in separate methods? put all these nested if statements in a separate method and just use the results of them here? it's pretty hard to keep track of the code flow rn

Reply:
I could put them a method, and I'll have to put that method call in an if statement. I did it this way so it doesn't change the job at all of single column folios. I can get rid of the if statements but that would require refactoring the job at least a bit